### ISSUE TITLE
Can't get example to work

### ISSUE THREAD

**Issue by welchwilmerck** [ansible/ansible-container#234]
[build-err.txt](https://github.com/ansible/ansible-container/files/475152/build-err.txt)
[build-out.txt](https://github.com/ansible/ansible-container/files/475153/build-out.txt)

##### ISSUE TYPE

 - Bug Report

##### container.yml 

```
example/ansible/container.yml from source
```

##### main.yml 

```
example/ansible/main.yml from source
```
##### OS / ENVIRONMENT

```
sudo ansible-container --debug version
Ansible Container, version 0.1.0
Linux, ip-54-40-187-6, 3.10.0-327.el7.x86_64, #1 SMP Thu Nov 19 22:10:57 UTC 2015, x86_64
2.7.5 (default, Aug 18 2016, 15:58:25)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-4)] /usr/bin/python
No DOCKER_HOST environment variable found. Assuming UNIX socket at /var/run/docker.sock
{u'Architecture': u'x86_64',
 u'BridgeNfIp6tables': False,
 u'BridgeNfIptables': False,
 u'CPUSet': True,
 u'CPUShares': True,
 u'CgroupDriver': u'cgroupfs',
 u'ClusterAdvertise': u'',
 u'ClusterStore': u'',
 u'Containers': 4,
 u'ContainersPaused': 0,
 u'ContainersRunning': 0,
 u'ContainersStopped': 4,
 u'CpuCfsPeriod': True,
 u'CpuCfsQuota': True,
 u'Debug': False,
 u'DockerRootDir': u'/var/lib/docker',
 u'Driver': u'devicemapper',
 u'DriverStatus': [[u'Pool Name', u'docker-202:1-9521344-pool'],
                   [u'Pool Blocksize', u'65.54 kB'],
                   [u'Base Device Size', u'10.74 GB'],
                   [u'Backing Filesystem', u'xfs'],
                   [u'Data file', u'/dev/loop0'],
                   [u'Metadata file', u'/dev/loop1'],
                   [u'Data Space Used', u'2.553 GB'],
                   [u'Data Space Total', u'107.4 GB'],
                   [u'Data Space Available', u'2.299 GB'],
                   [u'Metadata Space Used', u'3.727 MB'],
                   [u'Metadata Space Total', u'2.147 GB'],
                   [u'Metadata Space Available', u'2.144 GB'],
                   [u'Udev Sync Supported', u'true'],
                   [u'Deferred Removal Enabled', u'false'],
                   [u'Deferred Deletion Enabled', u'false'],
                   [u'Deferred Deleted Device Count', u'0'],
                   [u'Data loop file',
                    u'/var/lib/docker/devicemapper/devicemapper/data'],
                   [u'Metadata loop file',
                    u'/var/lib/docker/devicemapper/devicemapper/metadata'],
                   [u'Library Version', u'1.02.107-RHEL7 (2016-06-09)']],
 u'ExecutionDriver': u'',
 u'ExperimentalBuild': False,
 u'HttpProxy': u'',
 u'HttpsProxy': u'',
 u'ID': u'J5Z2:7RH4:BPLL:2RMV:HHHR:7RGM:R6WS:3VI3:FWGA:YFJI:GAWM:AZ7P',
 u'IPv4Forwarding': True,
 u'Images': 3,
 u'IndexServerAddress': u'https://index.docker.io/v1/',
 u'KernelMemory': True,
 u'KernelVersion': u'3.10.0-327.el7.x86_64',
 u'Labels': None,
 u'LoggingDriver': u'json-file',
 u'MemTotal': 3705864192,
 u'MemoryLimit': True,
 u'NCPU': 2,
 u'NEventsListener': 0,
 u'NFd': 20,
 u'NGoroutines': 32,
 u'Name': u'ip-54-40-187-6',
 u'NoProxy': u'',
 u'OSType': u'linux',
 u'OomKillDisable': True,
 u'OperatingSystem': u'CentOS Linux 7 (Core)',
 u'Plugins': {u'Authorization': None,
              u'Network': [u'null', u'host', u'bridge'],
              u'Volume': [u'local']},
 u'RegistryConfig': {u'IndexConfigs': {u'docker.io': {u'Mirrors': None,
                                                      u'Name': u'docker.io',
                                                      u'Official': True,
                                                      u'Secure': True}},
                     u'InsecureRegistryCIDRs': [u'127.0.0.0/8'],
                     u'Mirrors': None},
 u'ServerVersion': u'1.11.2',
 u'SwapLimit': True,
 u'SystemStatus': None,
 u'SystemTime': u'2016-09-14T22:57:20.706667522Z'}
{u'ApiVersion': u'1.23',
 u'Arch': u'amd64',
 u'BuildTime': u'2016-06-01T21:23:11.279703501+00:00',
 u'GitCommit': u'b9f10c9',
 u'GoVersion': u'go1.5.4',
 u'KernelVersion': u'3.10.0-327.el7.x86_64',
 u'Os': u'linux',
 u'Version': u'1.11.2'}
```
##### SUMMARY

Just validating the project and the example doesn't work.

##### STEPS TO REPRODUCE

Spin up a centos 7 instance (t2.medium, from ami-83bdf7e9)
Add docker repo
`yum install docker-engine-1.11.2 python-pip git`
`pip install --upgrade setuptools`
`pip install ansible-container`
start docker
clone the example directory and cd to it
`ansible-container build`
(sudo where necessary - didn't use venv)

##### ACTUAL RESULTS

This seems to be germane:
```
ansible-container_1  | fatal: [gulp]: UNREACHABLE! => {"changed": false, "msg": "Authentication or permission failure. In some cases, you may have been able to authenticate
and did not have permissions on the remote directory. Consider changing the remote temp path in ansible.cfg to a path rooted in \"/tmp\". Failed command was: ( umask 77 && m
kdir -p \"` echo $HOME/.ansible/tmp/ansible-tmp-1473889080.15-157810196845426 `\" && echo ansible-tmp-1473889080.15-157810196845426=\"` echo $HOME/.ansible/tmp/ansible-tmp-1
473889080.15-157810196845426 `\" ), exited with result 1", "unreachable": true}
```
```
ansible-container_1  |
ansible-container_1  | PLAY [gulp] ********************************************************************
ansible-container_1  |
ansible-container_1  | PLAY [static] ******************************************************************
ansible-container_1  |  to retry, use: --limit @main.retry
ansible-container_1  |
ansible-container_1  | PLAY RECAP *********************************************************************
ansible-container_1  | django                     : ok=15   changed=11   unreachable=0    failed=0
ansible-container_1  | gulp                       : ok=0    changed=0    unreachable=1    failed=0
ansible-container_1  | static                     : ok=0    changed=0    unreachable=1    failed=0
ansible-container_1  |
ansible_ansible-container_1 exited with code 3
Aborting on container exit...
Stopping ansible_static_1 ... done
Stopping ansible_gulp_1 ... done
Stopping ansible_django_1 ... done
Stopping ansible_postgresql_1 ... done
Ansible playbook run failed.
```

**Comment by chouseknecht** [comment]
I'll see if I can reproduce this today and post an update. Apologies for the inconvenience.

**Comment by j00bar** [comment]
Thank you for this report - would you mind running `ansible-container --debug build` so that we can see the stderr at the point of failure? Thanks! @chouseknecht I'm going to update the issue template to request the debug build output.

**Comment by welchwilmerck** [comment]
Added stdout and stderr.

**Comment by chouseknecht** [comment]
I'm running into the same thing. There seems to be two issues: 

- If the service config in `container.yml`  includes `user:`, we seem to be trying execute the build as that user. The build must run as root, because most likely no users exist yet.
- There's a race condition between the containers actually being ready to accept a connection via `docker exec` and playbook execution starting.

**Comment by chouseknecht** [comment]
Turns out it's purely a race condition. If I add the following play at the beginning of `main.yml`, it works perfectly:
```
- hosts: localhost
  gather_facts: no
  connection: local
  tasks:
    - name: Wait for 5 sec
      shell: sleep 5
```

**Comment by chouseknecht** [comment]
Added a step in the builder.sh script that causes the build container to wait for the other containers to be in a running state before starting playbook execution. To pick up the the change, you will need get latest build container image ansible/ansible-container-builder:0.2, and run the latest Ansible Container source. The following steps should do the trick:

- Install Ansible Container 0.2.0: `pip install --upgrade git+https://github.com/ansible/ansible-container.git`
- `docker rmi --force ansible/asible-container-builder:0.2` to remove the builder image from your image cache, if it exists.
- From your local copy of the Ansible Container project, cd into the `example` directory and start the build: `ansible-container build`

I'm noticing too on Centos:7 using Docker 1.11.2 that building the example app seems to hang at various points. It's not consistent, and I have no idea what causes it. However, upgrading to Docker 1.12.1 fixes it. I'm able to consistently build the project without any errors or hanging.

**Comment by welchwilmerck** [comment]
Thanks! I now have images. Of course, that just opens the way to the next problem. 'ansible-container run' is giving me:

The PWD variable is not set. Defaulting to a blank string.

ERROR: for django  create .: "." includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed.

I'll do more investigation before opening an issue, however.

**Comment by chouseknecht** [comment]
Hmm... 

I'm not seeing the PWD error.  Try running with `ansible-container --debug run`, so we can see if there is something else going on.

Also, if the build was successful, you should see the following images:

```
example-django                      20160917132442      3e376d508452        36 hours ago        1.038 GB
example-django                      latest              3e376d508452        36 hours ago        1.038 GB
example-static                      20160917132442      9b8f50e40dc4        36 hours ago        1.017 GB
example-static                      latest              9b8f50e40dc4        36 hours ago        1.017 GB
example-gulp                        20160917132442      554efe396b4b        36 hours ago        347.4 MB
example-gulp                        latest              554efe396b4b        36 hours ago        347.4 MB
```
